File: Assets/AltUnityTester/AltUnityDriver/Commands/InputActions/MoveMouseAndWait.cs
using Assets.AltUnityTester.AltUnityDriver.UnityStruct;

public class MoveMouseAndWait : AltBaseCommand
{
    Vector2 location;
    float duration;
    public MoveMouseAndWait(SocketSettings socketSettings, Vector2 location, float duration) : base(socketSettings)
    {
        this.location = location;
        this.duration = duration;
    }
    public void Execute(){
        new MoveMouse(SocketSettings,location, duration).Execute();
        System.Threading.Thread.Sleep((int)duration * 1000);
        string data;
        do
        {
            Socket.Client.Send(toBytes(CreateCommand("actionFinished")));
            data = Recvall();
        } while (data == "No");
        if (data.Equals("Yes"))
            return;
        HandleErrors(data);
    }
}

File: Assets/AltUnityTester-BindingsAndExamples(can_be_deleted)/Bindings~/python/altunityrunner/commands/InputActions/move_mouse_and_wait.py
from altunityrunner.commands.base_command import BaseCommand
from altunityrunner.commands.InputActions.move_mouse import MoveMouse
import time
class MoveMouseAndWait(BaseCommand):
    def __init__(self, socket,request_separator,request_end, x, y, duration):
        super().__init__(socket,request_separator,request_end)
        self.x=x
        self.y=y
        self.duration=duration
    
    def execute(self):
        data = MoveMouse(self.socket,self.request_separator,self.request_end,self.x, self.y, self.duration).execute()
        self.handle_errors(data)
        print('Wait for move mouse to finish')
        time.sleep(self.duration)
        action_in_progress = True
        while action_in_progress:
            action_finished = self.send_data(self.create_command('actionFinished'))
            self.handle_errors(action_finished)
            if action_finished is 'Yes':
                break
            elif action_finished != 'No':
                action_in_progress = False
        return self.handle_errors(data)